### DISCUSSION TITLE
Environment Variables aren't processed if running Docker binary normally

### DISCUSSION THREAD

**Discussion by DUOLabs333** [dani-garcia/vaultwarden#2645] [Q&A] [answered]
### Subject of the issue
If I run `vaultwarden/server:latest` with `source /etc/.env; /start.sh` or `source /etc/.etc; /vaultwarden`, the new environment variables aren't used (`DATA_FOLDER` is still `/data`, and it is using a sqlite database, not my postgres one.

### Deployment environment

* vaultwarden version: 1.25.1

* Install method: Docker image, but run manually in a chroot (the compiled version --- it's too old to show the version number at start-up --- does not have this problem).

* Clients used: 

* Reverse proxy and version: 

* MySQL/MariaDB or PostgreSQL version: 

* Other relevant details:

### Steps to reproduce
1. Download the tar ball of the docker image
2. Chroot into it and make an `/etc/.env` file with `DATA_FOLDER=/etc/data`.
3. Copy `/data` to `/etc/data`.
4. Run `source /etc/.env; /start.sh`.
5. See that the data folder used is not  `/etc/data`

### Expected behaviour
The variables in `/etc/.env` are not respected.

**Comment by BlackDex** [comment]
I'm unsure what you try to do here.
If  you want to use a `.env` file for vaultwarden to read, either mount/put that `.env` file in the same folder as the `vaultwarden` binary. Or set the `ENV_FILE=/path/to/env-file` before you start Vaultwarden.

Also, a `source /etc/.env ; my_script.sh` doesn't provide those env's to `my_script.sh` unless you have `export VAR_NAME` inside the `.env` file. That just not how bash or sh work, they do not have access to non-exported variables.


Also, all version of Vaultwarden/Bitwarden_RS show the version numbers during startup, and else it should be somewhere in the docker image metadata.

**Comment by DUOLabs333** [comment]
I tried `set -a; source /etc/.env; /vaultwarden`, which should export all variables, but that didn't work either.

**Comment by DUOLabs333** [comment]
`ENV_FILE` works though.

**Comment by BlackDex** [comment]
Not sure why `set -a` doesn't work though.
Could be the environment, could be something strange during export of all the vars compared to Vaultwarden reading the file it self. Could be the shell environment bash vs sh, or maybe even the bash/sh version it self.

Main thing is, Vaultwarden is able to read exported environment variables and read the `.env` file.
So, I think there is no issue.

**Comment by DUOLabs333** [comment]
The problem is, I want the .env file to override any of the default environment variables, which is why I tried exporting them.

**Reply by BlackDex** [reply]
Not sure how overwriting works and if that works correctly.

also keep in mind that if there is a `config.json`, so, anything saved via the admin interface will always override all the envs.

**Comment by DUOLabs333** [comment, marked as answer]
Also, found the problem: I wrote `source` instead of `.`, as the shell is `/bin/sh`.
